src/module_wrap.cc

```diff
@@ -89,68 +91,64 @@ void ModuleWrap::New(const FunctionCallbackInfo<Value>& args) {
 
   Local<String> url = args[1].As<String>();
 
-  Local<Module> mod;
+  Local<Module> module;
 
   // compile
   {
     ScriptOrigin origin(url,
-                        Integer::New(iso, 0),
-                        Integer::New(iso, 0),
-                        False(iso),
-                        Integer::New(iso, 0),
-                        FIXED_ONE_BYTE_STRING(iso, ""),
-                        False(iso),
-                        False(iso),
-                        True(iso));
+                        Integer::New(isolate, 0),             // line offset
+                        Integer::New(isolate, 0),             // column offset
+                        False(isolate),                       // is cross origin
+                        Local<Integer>(),                     // script id
+                        Local<Value>(),                       // source map URL
+                        False(isolate),                       // is opaque (?)
+                        False(isolate),                       // is WASM
+                        True(isolate));                       // is ES6 module
     ScriptCompiler::Source source(source_text, origin);
-    auto maybe_mod = ScriptCompiler::CompileModule(iso, &source);
-    if (maybe_mod.IsEmpty()) {
-        return;
-    }
-    mod = maybe_mod.ToLocalChecked();
+    if (!ScriptCompiler::CompileModule(isolate, &source).ToLocal(&module))
+      return;
   }
 
-  auto that = args.This();
-  auto ctx = that->CreationContext();
-  auto url_str = FIXED_ONE_BYTE_STRING(iso, "url");
+  Local<Object> that = args.This();
+  Local<Context> context = that->CreationContext();
+  Local<String> url_str = FIXED_ONE_BYTE_STRING(isolate, "url");
 
-  if (!that->Set(ctx, url_str, url).FromMaybe(false)) {
+  if (!that->Set(context, url_str, url).FromMaybe(false)) {
     return;
   }
 
-  ModuleWrap* obj =
-      new ModuleWrap(Environment::GetCurrent(ctx), that, mod, url);
+  ModuleWrap* obj = new ModuleWrap(env, that, module, url);
 
-  env->module_map.emplace(mod->GetIdentityHash(), obj);
+  env->module_map.emplace(module->GetIdentityHash(), obj);
   Wrap(that, obj);
 
-  that->SetIntegrityLevel(ctx, IntegrityLevel::kFrozen);
+  that->SetIntegrityLevel(context, IntegrityLevel::kFrozen);
   args.GetReturnValue().Set(that);
 }
 
 void ModuleWrap::Link(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  Isolate* iso = args.GetIsolate();
-  EscapableHandleScope handle_scope(iso);
+  Isolate* isolate = args.GetIsolate();
   if (!args[0]->IsFunction()) {
     env->ThrowError("first argument is not a function");
     return;
   }
 
   Local<Function> resolver_arg = args[0].As<Function>();
 
-  auto that = args.This();
+  Local<Object> that = args.This();
   ModuleWrap* obj = Unwrap<ModuleWrap>(that);
+  CHECK_NE(obj, nullptr);
   auto mod_context = that->CreationContext();
```

Comment:
don't we have an appropriate Local abstraction for this `auto`?

Reply:
Yup, thanks for pointing it out :)